src/Infrastructure.VS/WorkspaceService.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.Infrastructure.VS
+{
+    public interface IWorkspaceService
+    {
+        /// <summary>
+        /// Returns the root directory for projects.
+        /// Will return null if the root directory could not be retrieved,
```

Comment:
The description isn't very clear - what does "_root directory for projects_" mean? A project does not have to under the solution folder.

Is the following description accurate?

```
<summary>
Returns the current solution directory, or null if there is not an open solution.
</summary>
<remarks>
For "normal" solutions, returns the directory containing the solution file.
For folder workspaces, returns the path to the folder that was opened.
</remarks>
```